cedar-policy-core/src/ast/name.rs

```diff
@@ -307,6 +312,19 @@ impl SlotId {
     pub fn is_resource(&self) -> bool {
         matches!(self, Self(ValidSlotId::Resource))
     }
+
+    /// Check if a slot represents a generalized slot
+    pub fn is_generalized_slot(&self) -> bool {
+        matches!(self, Self(ValidSlotId::GeneralizedSlot(_)))
+    }
+
+    /// Returns the id if the slot is a `generalized slot`
+    pub fn extract_id_out_of_generalized_slot(&self) -> Option<Id> {
+        match self {
+            Self(ValidSlotId::GeneralizedSlot(id)) => Some(id.clone()),
+            _ => None,
+        }
+    }
```

Comment:
I suggest returning `Option<&Id>` from this function, to avoid the clone.  (Caller can always clone if they need an owned `Id`.)